coin_chk_libhdr.m4

```diff
@@ -0,0 +1,432 @@
+# is deliberate.  There's no guarantee that the specified library can be
+# accessed just yet with the specified flags. Put another way, unless the
+# user requests a compile and/or link check, all we're doing here is filling
+# in variables using a complicated algorithm.
+
+AC_DEFUN([AC_COIN_FIND_PRIM_LIBHDR],
+[
+dnl Set default values for flags, action, and status. These are taken from
+dnl the macro parameters, if given. Otherwise, make something up.
+
+  m4_tolower($1_lflags)="m4_default([$2],[-l$1])"
+  m4_tolower($1_cflags)="m4_default([$3],[])"
+  m4_tolower($1_data)="m4_default([$4],[/usr/local/share/$1])"
+  m4_if(AC_COIN_LIBHDR_DFLT_HACK([usage],[$7],[yes],[foo]),yes,
+    [m4_tolower(coin_has_$1)=requested],
+    [m4_tolower(coin_has_$1)=skipping
+     m4_tolower($1_failmode)='default'])
+
+dnl See if the user specified --with-prim.  If the value is something other
+dnl than 'yes' or 'no' and the client specified dataonly, the value is assigned
+dnl to prim_data, otherwise to prim_lflags.
+
+  withval="$m4_tolower(with_$1)"
+  if test -n "$withval" ; then
+    case "$withval" in
+      no )
+        m4_tolower(coin_has_$1)=skipping
+	m4_tolower($1_failmode)='command line'
+        ;;
+      yes )
+        m4_tolower(coin_has_$1)=requested
+	m4_tolower($1_failmode)=''
+        ;;
+      * )
+        m4_tolower(coin_has_$1)=requested
+	m4_tolower($1_failmode)=''
+        m4_if(m4_default($8,nodata),dataonly,
+          [m4_tolower($1_data)="$withval"],
+          [m4_tolower($1_lflags)="$withval"])
+        ;;
+    esac
+  fi
+
+dnl As long as we're not dataonly and we're not skipping prim, check for
+dnl --with-prim-lflags and --with-prim-cflgs. Values will override parameter
+dnl values.
+
+  m4_if(m4_default($8,nodata),dataonly,[],
+    [if test "$m4_tolower(coin_has_$1)" != skipping ; then
+       withval="$m4_tolower(with_$1_lflags)"
+       if test -n "$withval" ; then
+	 m4_tolower(coin_has_$1)=requested
+	 m4_tolower($1_lflags)="$withval"
+       fi
+
+       withval="$m4_tolower(with_$1_cflags)"
+       if test -n "$withval" ; then
+	 m4_tolower(coin_has_$1)=requested
+	 m4_tolower($1_cflags)="$withval"
+       fi
+     fi])
+
+dnl If we're not nodata and we're not skipping prim, check for
+dnl --with-prim-data. A value will override the parameter value.
+
+  m4_if(m4_default($8,nodata),nodata,[],
+    [if test "$m4_tolower(coin_has_$1)" != skipping ; then
+       withval="$m4_tolower(with_$1_data)"
+       if test -n "$withval" ; then
+	 m4_tolower(coin_has_$1)=requested
+	 m4_tolower($1_data)="$withval"
+       fi
+     fi])
+
+dnl At this point, coin_has_prim can be one of skipping (user said no, or
+dnl default was no without override), or requested (user said yes, or default
+dnl was yes without override).
+
+dnl If we have [includes], try to compile them.
+
+  m4_ifnblank([$6],
+    [if test $m4_tolower(coin_has_$1) != skipping ; then
+       ac_save_CXXFLAGS=$CXXFLAGS
+       CXXFLAGS="$m4_tolower($1_cflags)"
```

Comment:
Shouldn't this be `CPPFLAGS`?

The user may have specified `AC_LANG_PUSH(C)` and that C++ flags would be ignored?

Reply:
Yes, CPPFLAGS seems to be the safe choice to me.

Reply:
Yep. Fixed.